### DISCUSSION TITLE
Best practice for database clients and calls?

### DISCUSSION THREAD

**Discussion by GionRubitschung** [nuxt/nuxt#16391] [Questions] [answered]
Hello Everyone

I am using prisma to connect to my mongo database. Only the api is allowed to access the databse, not the frontend. Since I have some db calls that I need more than once, I moved them into composables, for example `getEntities` or `getEntityById` which I then can call under these names in the server routes.

Nuxt feels very functional to me and not object oriented. I would like to keep this functional approach through my application so that I don't end up with service classes like `EntityService` which then access repository classes like `EntityRepository`.
This raises a problem. In each composable that is a db call I initialize the prisma client. This doesn't feel best practice to me.

When the app eventually goes live it will have an estimated traffic of about 1000 to 5000 users daily and each one of these users will make calls to the api which calls the db. In my logic this means that on each call the db client gets newly initialized or in other words a new connection gets opened. This could also be a security issue for us.

So my question is, how can I handle something like this? Is the db calls inside composables the best way?

**Comment by fathi91** [comment, marked as answer]
Hi,
you should only have one instance of PrismaClient and re-use it across your application,
create a file in for example `/server/utils/db-client.ts`
that has the following:
```javascript
import { PrismaClient /*Prisma*/ } from "@prisma/client";
const dbClient = new PrismaClient({
  //log: ["query"],
});
export { dbClient /*Prisma */ };
```
use it to initialize the PrismaClient and you can also add middleware here as well if you are familiar with prisma middleware
then import it where you want to call the database

that way you'd have only one instance of the client.

you can read more about connection management in prisma documentation: https://www.prisma.io/docs/guides/performance-and-optimization/connection-management

**Reply by GionRubitschung** [reply]
Thank you very much! Exactly what I needed.
